File: group3_miniproject_fullstack/miniproject_spring/src/main/java/com/example/miniproject_spring/service/CustomerService.java
package com.example.miniproject_spring.service;

import java.util.List;

import org.springframework.stereotype.Service;

import com.example.miniproject_spring.model.Customers;
import com.example.miniproject_spring.model.Group3DAO;

@Service
public class CustomerService {
    Group3DAO group3DAO = new Group3DAO();

    public List<Customers> getAllCustomers(){
        return group3DAO.getStatementResultList("SELECT * FROM CUSTOMERS",Customers.class);
    }

    public Customers getCustomerById(int id){
        return group3DAO.getPreparedStatementSingleResultList("SELECT * FROM CUSTOMERS WHERE id = ?",Customers.class, id);
    }

    public Long getCustomerCount(){
        return getAllCustomers().stream().count();
    }

    
}

File: project2_React_Springboot_Integration/group3_miniproject_react/src/service/CustomerService.js
import axios from "axios";

const REST_API_BASE_URL = "http://localhost:8080/api/customers";

export const getAllCustomers = () => axios.get(REST_API_BASE_URL);

export const getCustomerCount = (customerId) =>
	axios.get(REST_API_BASE_URL + "/count");

export const getCustomerById = (customerId) =>
	axios.get(REST_API_BASE_URL + "/" + customerId);
